cmd/ocm-backplane/cloud/common.go

```diff
@@ -263,17 +267,24 @@ func (cfg *QueryConfig) getIsolatedCredentials(ocmToken string) (aws.Credentials
 		return aws.Credentials{}, fmt.Errorf("failed to unmarshal response: %w", err)
 	}
 
-	roleAssumeSequence := make([]string, 0, len(roleChainResponse.AssumptionSequence))
-	for _, namedRoleArn := range roleChainResponse.AssumptionSequence {
-		roleAssumeSequence = append(roleAssumeSequence, namedRoleArn.Arn)
+	assumeRoleArnSessionSequence := make([]awsutil.RoleArnSession, 0, len(roleChainResponse.AssumptionSequence))
+	for _, namedRoleArnEntry := range roleChainResponse.AssumptionSequence {
+		roleArnSession := awsutil.RoleArnSession{RoleArn: namedRoleArnEntry.Arn}
+		if namedRoleArnEntry.Name == CustomerRoleArnName {
+			data := []byte(email)
+			roleArnSession.RoleSessionName = fmt.Sprintf("%x", md5.Sum(data)) //nolint:gosec
```

Comment:
It's definitely possible - it comes down to how quickly we want this out the door.

To go this route, we would add a new API endpoint that takes a string and returns an md5 hash, using the same existing functions that are used in the API today for performing that conversion. Then the bp cli would call that endpoint instead.

I'm happy to go that route, and then we can keep this PR in a "hold" until that API change gets released. Thoughts?

Reply:
Actually another thought - since this is only for the new flow, we could update the response object of the existing API call to include a new property for the role session name to be used for the customer's support role. That would be substantially less overhead and allow us quicker turnover.